test/openmp-mpi-cpp/downstream/CMakeLists.txt

```
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.14)
+
+project(openmp-cpp-fortran-user LANGUAGES CXX)
+
+set(BLT_CXX_STD c++14 CACHE STRING "")
+set(ENABLE_MPI On CACHE BOOL "")
+
+if (NOT BLT_LOADED)
+    if (DEFINED BLT_SOURCE_DIR)
+      if (NOT EXISTS ${BLT_SOURCE_DIR}/SetupBLT.cmake)
+        message(FATAL_ERROR "Given BLT_SOURCE_DIR does not contain SetupBLT.cmake")
+      endif()
+    else ()
+      set (BLT_SOURCE_DIR ${PROJECT_SOURCE_DIR}/../../../blt CACHE PATH "")
+  
+      if (NOT EXISTS ${BLT_SOURCE_DIR}/SetupBLT.cmake)
+        message(FATAL_ERROR "\
+        The BLT submodule is not present. \
+        If in git repository run the following two commands:\n \
+        git submodule init\n \
+        git submodule update")
+      endif ()
+    endif ()
+  
+    include(${BLT_SOURCE_DIR}/SetupBLT.cmake)
+endif()
```

Suggestion:
Do we need the flexibility of supporting an external BLT? or can we just assume that the examples in this repo are using its blt submodule? If it's the latter, we might consider simplifying this repeated block to something like:
```suggestion
 set (BLT_SOURCE_DIR ${PROJECT_SOURCE_DIR}/../../../blt CACHE PATH "")
 if (NOT EXISTS ${BLT_SOURCE_DIR}/SetupBLT.cmake)
     message(FATAL_ERROR "The BLT submodule must be initialized")
endif ()
include(${BLT_SOURCE_DIR}/SetupBLT.cmake)
```

---

moved this logic into a `.cmake` in the `test/cmake` directory